### ISSUE TITLE
Donation button is not working

### ISSUE THREAD

**Issue by SaurabhS55** [Innovateninjas/Paws-frontend#304]
Hey @codewarnab,  I've checked out your website, and I must say, it's impressive! I appreciate the effort you've put into it. However, I've noticed a bug in donation page. Plzz correct me if I may getting wrong, but upon clicking donate button or any prior buttons of amount, the payment interface is not opening. 

screenshots: 
![image](https://github.com/Innovateninjas/Paws-frontend/assets/119050714/74fe633e-2489-4fbe-a670-d0f79f8cd3e6)

I think there might be some sort of CORS issue or request header issues.
So, Can we use payment gateways like stripe or razourpay.
I would like to work on this issue. Could you assign this issue to me under GSSOC'24 label?

**Comment by codewarnab** [comment]
yes we are aware of that issue link doesn't opens in desktop but in phone redirects to UPI app 
you can intregate razorpay or stripe it will ask for documents i guess ? we didn't explore those alternatives yet if you want you can work on it( comment yes then i will assign you ) after you have integrated the payment gateway console log the payment details after successful payment i will create a separate issue for sending them to backed and storing them

**Comment by SaurabhS55** [comment]
Yes

**Comment by codewarnab** [comment]
okay best of luck also try to avoid conflicts because someone already working on this page #294 you guys may coordinate with each other in discussion

**Comment by codewarnab** [comment]
@SaurabhS55 whats the update its been a week

**Comment by SaurabhS55** [comment]
Actually, last week I had my semester exams. 
Now, I've started working on it. I just want to ask you, which branch is deployed on production? Is it the main branch or another one? As donate button on footer is missing in main branch code.

![image](https://github.com/Innovateninjas/Paws-frontend/assets/119050714/a46f271f-b6b6-42ad-9c5e-ca6a4eeeef82)

**Comment by SaurabhS55** [comment]
Sure we can

**Comment by codewarnab** [comment]
> Actually, last week I had my semester exams. Now, I've started working on it. I just want to ask you, which branch is deployed on production? Is it the main branch or another one? As donate button on footer is missing in main branch code.
> 
> ![image](https://private-user-images.githubusercontent.com/119050714/331860974-a46f271f-b6b6-42ad-9c5e-ca6a4eeeef82.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTY2MDcwMDIsIm5iZiI6MTcxNjYwNjcwMiwicGF0aCI6Ii8xMTkwNTA3MTQvMzMxODYwOTc0LWE0NmYyNzFmLWI2YjYtNDJhZC05YzVlLWNhNmE0ZWVlZWY4Mi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNTI1JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDUyNVQwMzExNDJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iOGMwOTJmZTBlMzIwNWRiYmQ1ZTA3YTE1ZGVhODdkNGQxNDgwMDhlZWNmM2E4N2Q4NzJmYzVlYWRiMGE5ODNmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.mlGogibdeYmLBChDUuj6H2G4TcIwELdXQR3XABliHXw)

after login you can see the donation tab

**Comment by codewarnab** [comment]
@SaurabhS55 whats the update of the payment gateway ?

**Comment by SaurabhS55** [comment]
Ok, till now I tried stripe and razourpay, 
But from Front-end it is only possible to open their payment interface where user can click on donate button and donate money. But donate 100 rs, 200 rs or 500 rs button will not work as we can't call payment intent from Front-end or if we call it will be vulnerable. So, setting money values could be possible from backend only.

**Comment by SaurabhS55** [comment]
For testing purpose should I continue with only donate button where payment interface will work independent of app. We just redirecting from donate button to interface for payment. If yes, I will do it within a day and make pr of it for review.

**Comment by codewarnab** [comment]
> For testing purpose should I continue with only donate button where payment interface will work independent of app. We just redirecting from donate button to interface for payment. If yes, I will do it within a day and make pr of it for review.

Yup okay , take your time

**Comment by SaurabhS55** [comment]
Hey @codewarnab, I just have made pr of this issue, review it once.
